### PR TITLE
Scheduled job for question ,question set and content

### PR DESCRIPTION
**Pull request by Novfal-202**
Job process for bulk upload question ,question set and content completed.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    app.use(amlErrorHandler);
+
+    // 404 handler for unknown API requests
+    app.use((req: Request, res: Response, next: NextFunction) => {
+      next({ statusCode: NOT_FOUND, message: 'Not found' });
+    });
+
+    // Database connection
+    await AppDataSource.sync()
+      .then(() => logger.info('Database connection successful'))
```
Comment (Hari-1027) [inline src/app.ts:58]: please remove this

File: src/config/config.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    name: get(envVars, 'LOG_FILE_NAME', 'AML-log'),
+    size: get(envVars, 'LOG_FILE_SIZE', '20m'),
+    zippedArchive: isEqual(get(envVars, 'LOG_FILE_ZIP_ARCHIVE', 'false'), 'true'),
+  },
+  envPort: get(envVars, 'AML_SERVICE_APPLICATION_PORT', '4000'),
+  applicationEnv: get(envVars, 'AML_SERVICE_APPLICATION_ENV', 'development'),
+  appVersion: get(envVars, 'AML_SERVICE_APP_VERSION', '1.1'),
+  DB: {
+    host: get(envVars, 'AML_SERVICE_DB_HOST', 'localhost'),
+    port: Number(get(envVars, 'AML_SERVICE_DB_PORT')),
```
Comment (Hari-1027) [inline src/config/config.ts:20]: add alternative port

File: src/app.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      .catch((err: any) => logger.error('Error in database connection', { message: err.message }));
```
Comment (Hari-1027) [inline src/app.ts:59]: please compare the main repo and correct all common comments.

File: src/controllers/BulkUpload/questionStatusJob.ts [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+import { CronJob } from 'cron';
+import { getAllCloudFolder, getQuestionSignedUrl, getTemplateSignedUrl } from '../../services/awsService';
+import { getProcessByMetaData, updateProcess } from '../../services/process';
+import path from 'path';
+import AdmZip from 'adm-zip';
+import { appConfiguration } from '../../config';
+
+const { fibCsvFileName, gridCsvFileName, mcqCsvFileName } = appConfiguration;
+
+const checkStatus = new CronJob('0 9,21 * * *', async () => {
```
Comment (Hari-1027) [inline src/controllers/BulkUpload/questionStatusJob.ts:12]: this is configurable. the cron timer should not be inside the controller

File: src/app.ts (hunk 1 above)
Comment (Novfal-202) [reply src/app.ts:58]: updated

File: src/app.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    // Start the server
+    app.listen(envPort, () => {
+      logger.info(`Listening on port.`);
```
[the remark this answers is not in the document]
Comment (Novfal-202) [reply src/app.ts:63]: updated

File: src/config/config.ts (hunk 2 above)
Comment (Novfal-202) [reply src/config/config.ts:20]: updated

File: src/app.ts (hunk 3 above)
Comment (Novfal-202) [reply src/app.ts:59]: changes completed

File: src/controllers/BulkUpload/questionStatusJob.ts (hunk 4 above)
Comment (Novfal-202) [reply src/controllers/BulkUpload/questionStatusJob.ts:12]: updated

File: src/controllers/BulkUpload/questionStatusJob.ts [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+  try {
+    const processInfo = await getProcessByMetaData({ status: 'open' });
+    const { getProcess } = processInfo;
+
+    let validFileNames: string[];
+
+    for (const process of getProcess) {
+      const { process_id, fileName, name } = process;
+      switch (name) {
+        case 'fib.zip':
```
[the remark this answers is not in the document]
Comment (Novfal-202) [reply src/controllers/BulkUpload/questionStatusJob.ts:22]: updated

File: .env.example [hunk 7]
```diff
@@ -0,0 +1,11 @@
+AML_SERVICE_APPLICATION_PORT = 3002
+AML_SERVICE_APPLICATION_ENV = "development" 
+AML_SERVICE_APP_VERSION="1.0"
+AML_SERVICE_DB_HOST=localhost
+AML_SERVICE_DB_PORT= 5432
+AML_SERVICE_DB_USER=postgres
+AML_SERVICE_DB_PASS=postgres
+AML_SERVICE_DB_NAME=postgres
+BUCKET_NAME=aml-dev-otsv83nane
```
Comment (HarishGangula) [inline .env.example:9]: Replace with dummy data

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,9 @@
+FROM --platform=linux/amd64 node:20-alpine
+RUN apk add --no-cache postgresql-client
```
Comment (HarishGangula) [inline Dockerfile:2]: Removed it

File: package.json [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    "lint": "eslint .",
+    "test": "mocha --exit",
+    "lint:fix": "eslint --fix .",
+    "prettier": "prettier --check .",
+    "prettier:fix": "prettier --write .",
+    "prepare": "husky install"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
Comment (HarishGangula) [inline package.json:26]: MIT

File: src/controllers/BulkUpload/questionStatusJob.ts [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+import path from 'path';
+import AdmZip from 'adm-zip';
+import { appConfiguration } from '../../config';
+
+const { csvFileName, cronJobPrcessUpdate } = appConfiguration;
+export const scheduleCronJob = () => {
+  const checkStatus = new CronJob(cronJobPrcessUpdate, async () => {
+    try {
+      const processInfo = await getProcessByMetaData({ status: 'open' });
+      const { getProcess } = processInfo;
```
Comment (HarishGangula) [inline src/controllers/BulkUpload/questionStatusJob.ts:15]: 1. Get the Zip file if not throw error
2. validate zip file contents if not throw error
3. validate the content in each file if not throw error
4. process the request 
5. insert to staging table
6. process each entity(question, question set, content) if not throw error
7. update the process status - successful

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: src/controllers/BulkUpload/questionStatusJob.ts (hunk 10 above)
Comment (Novfal-202) [reply src/controllers/BulkUpload/questionStatusJob.ts]: Upto second point in these PR .next point are merge once the initial PR merged.

File: package.json (hunk 9 above)
Comment (Novfal-202) [reply package.json]: updated.

File: Dockerfile (hunk 8 above)
Comment (Novfal-202) [reply Dockerfile]: updated

File: .env.example (hunk 7 above)
Comment (Novfal-202) [reply .env.example]: updated.

File: .env.example [hunk 11]
```diff
@@ -0,0 +1,12 @@
+AML_SERVICE_APPLICATION_PORT = 3002
+AML_SERVICE_APPLICATION_ENV = "development" 
+AML_SERVICE_APP_VERSION="1.0"
+AML_SERVICE_DB_HOST=localhost
+AML_SERVICE_DB_PORT= 5432
+AML_SERVICE_DB_USER=postgres
+AML_SERVICE_DB_PASS=postgres
+AML_SERVICE_DB_NAME=bulk_upload_service
```
[the remark this answers is not in the document]
Comment (Novfal-202) [reply .env.example]: updated.

File: .env.example [hunk 12]
```diff
@@ -0,0 +1,12 @@
+AML_SERVICE_APPLICATION_PORT = 3002
+AML_SERVICE_APPLICATION_ENV = "development"
```
[the remark this answers is not in the document]
Comment (Novfal-202) [reply .env.example]: updated.

[the remark this answers is not in the document]
Comment (Novfal-202) [reply .env.example:2]: removed

File: .env.example (hunk 11 above)
[the remark this answers is not in the document]
Comment (Novfal-202) [reply .env.example]: updated.

File: .env.example (hunk 7 above)
Comment (Novfal-202) [reply .env.example]: bucket name updated as dummy Bucket

**Comment by Novfal-202** [comment]
As per mentioned changes and modification completed.

**Comment by HarishGangula** [comment]
{
error:  bulkUploadError
data: null
}


{
error:  null
data: {
isValid: true
}
}

{
error:  null
data: {
zipContents: []
}
}

**Comment by Novfal-202** [comment]
@HarishGangula PR updated with latest changes .

**Review by HarishGangula — APPROVED** [review]

**Comment by HarishGangula** [comment]
Merging with implementation, Need to test with Actual data

**Pull request merged**